app/src/main/java/com/kasiaak/kursakademiaandroida/core/base/UseCase.kt

```diff
@@ -0,0 +1,21 @@
+package com.kasiaak.kursakademiaandroida.core.base
+
+import kotlinx.coroutines.*
+
+abstract class UseCase<out Type, in Params> {
+    abstract suspend fun action(params: Params): Type
+
+    operator fun invoke(
+        params: Params,
+        scope: CoroutineScope,
+        executionDispatcher: CoroutineDispatcher = Dispatchers.IO,
+        onResult: (Result<Type>) -> Unit = {}
+    ) {
+        scope.launch {
+            val result = withContext(Dispatchers.IO) {
```

Comment:
Chociaż w większości przypadków będziemy korzystać właśnie z dispatchera IO, to chcemy sobie dać elastyczność i możliwość ewentualnej zmiany. Z tego względu przekazujemy `executionDispatcher` jako argument funkcji `invoke` i to właśnie on powinien zostać tutaj ustawiony. Jest on domyślnie ustawiony na `Dispatchers.IO` więc zasada działania będzie taka sama.